common/include/revalloc.h

```diff
@@ -109,55 +87,48 @@ class Allocator : public Policy {
   typedef typename AllocationPolicy::const_reference const_reference;
   typedef typename AllocationPolicy::value_type      value_type;
 
-public:
   template<typename U>
   struct rebind {
     typedef Allocator<U, typename AllocationPolicy::rebind<U>::other> other;
   };
 
-public:
-  inline explicit Allocator() {}
-
-  inline ~Allocator() {}
-
-  inline Allocator( Allocator const& rhs ) : Policy( rhs ) {}
-
-  template<typename U>
-  inline Allocator( Allocator<U> const& ) {}
+  explicit Allocator()          = default;
+  ~Allocator()                  = default;
+  Allocator( Allocator const& ) = default;
 
-  template<typename U, typename P>
-  inline Allocator( Allocator<U, P> const& rhs ) : Policy( rhs ) {}
+  template<typename... U>
+  Allocator( Allocator<U...> const& rhs ) : Policy( rhs ) {}
```

Comment:
Making this variadic is interesting - why just this constructor?

Reply:
> Making this variadic is interesting - why just this constructor?

The other constructors like copy constructor cannot be templated and still be considered "the" copy constructor. Some constructors need a very specific signature and cannot be templated.

Reply:
Also note that non-templated functions are always rated higher in priority than templated functions. If a function/constructor call can exactly match a non-templated function overload, then it takes priority over all templated overloads. C++ legalism.